DOCS-9876 sandboxes (#189)
mulesoft/docs-api-community-manager: modules/ROOT/pages/sandboxes.adoc

> = Sandboxes

Reviewer: Can we make this title more robust? What, specifically, ABOUT sandboxes that the user wants to know to achieve a benefit?

ifndef::env-site,env-github[]
include::_attributes.adoc[]
endif::[]
:imagesdir: ../assets/images/

After you install an Anypoint API Community Manager update, it is not possible to undo the installation
and return to a previous version. Follow the best-practice workflow by creating safe sandbox
environments and using them to test each update before installing the update in your production environment.

> Best-practice workflow:

Reviewer: Likewise, how about a sentence, something like "Follow this workflow to test an update before using it:"

Reviewer: Actually...this is a task, not a workflow (imperative voice). You might want to reformat this as a topic set:

= Create and Use Sandboxes as Safe Testing Environments
intro
== Sandbox Testing Workflow (although these are really use steps and probably should have a task title...which is complicated by the following subtopic being...:
== Create a Sandbox

> . Keep a partial copy sandbox or a full copy sandbox as a long-term staging environment.

Reviewer: I was not sure what you meant by "partial copy sandbox" and "full copy sandbox": copy of what? 
Do you actually mean "Keep either a partial or full copy of your data in a sandbox as a long-term staging environment."?

+

> The staging environment will have a partial or full copy of the data in the production environment.

Reviewer: I think you can omit this sentence if you make the previous one more clear.

. Create a short-term developer sandbox.
. Install the update in the developer sandbox and test it.
. Discard the developer sandbox.
. Install the update in the staging sandbox and test it.
. Install the update in your production environment.

To create a new sandbox:

> 

Reviewer: Because the following is a lot of steps, how about separating creating and validating into separate tasks?

. In *Setup*, search for *Sandboxes* using the *Quick Find* box and click *New Sandbox*.

> . Enter a name and description and create a new developer sandbox.

Reviewer: So...should the topic title include "Developer"?

Reviewer: Does entering a name and description automatically start sandbox creation? There is no "enter" or "ok" action?

> . After several minutes to a few hours, sandbox creation completes and sends you a notification email.

Reviewer: This is not a step; it is followup info to the previous step.

Reviewer: Also...if the user has to wait up to a few hours, is this really all one task?
Consider reevaluating your narrative: 
What is actually a workflow (flow of actions) vs. a task (providing user steps to complete those actions). 
Do we need to define different types of sandboxes? 
Are we assuming that some sandboxes already exist? 
Are *all* sandboxes developer sandboxes and, if so, why specify this in some places but not others?
Do we need to specify that you must be an admin to perform these tasks?

> . In *Setup*, search for *Sandboxes* using the *Quick Find* box and log in to your new sandbox.
> .. Note that the user name has the sandbox name as a suffix.
> .. Enter your administrator password.

Reviewer: This should be two steps: finding and logging in.
The first substep is not a step at all.
```suggestion
. In *Setup*, search for *Sandboxes* using the *Quick Find* box.
. Log in to your new sandbox using your user name, including the sandbox name as a suffix, and enter your administrator password.
```
Consider following this with a screenshot showing what this user name an password might look like.

> . Validate the sandbox:

Reviewer: See prev. comments about organization.

.. Open the application launcher.
.. Open the *API Community Manager* application.
.. Publish a few APIs.
.. Open the builder.
.. Publish your community.
.. Open your community's live site.

> .. Validate the site is operational.

Reviewer: This task is problematic. 
We have gone from steps that create a staging environment,
backtracked to creating a new sandbox,
and this validation task doesn't tell you HOW to validate the sandbox but rather the flow from publishing APIs and the community, to making sure that the community site is working????
